### PR TITLE
Use `googlesitekit-data` imports directly instead of object destructing.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

This refactors our data imports to not require the entire registry to be imported and destructured.

Addresses issue:

- #8769

## Relevant technical choices

The reason we have imported our data module as a single object and then destructed values from it, eg:

```jsx
import Data from 'googlesitekit-data';

const { useDispatch, useSelect } = Data;
```

instead of

```jsx
import { useDispatch, useSelect } from 'googlesitekit-data';
```

isn't intentional. It's just a result of us not optimising our webpack/Jest configs at the time we created these modules. It ended up ballooning and just being an "accepted" thing, but it's ergonomically a bit awkward that we can't directly import these modules.

It's also annoying that you can't option + click in VSCode to "follow" the imports, like you can with most modules.

This PR solves both issues. It's obviously a huge diff because it modifies every file that imports from `googlesitekit-data`.

But I think this is a cleaner, clearer, more straightforward way to use these files going forward 🙂 

It seems to pass all tests/lints. I think there might be some storybook import fixes needed, but it's largely good-to-go. 👍🏻 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/data/index.js
Comment (tofumatt) [inline assets/js/googlesitekit/data/index.js:56]: Source: https://github.com/google/site-kit-wp/releases/tag/1.3.0

**Review by zutigrm — APPROVED** [review]
Thanks @tofumatt nice one! This is pretty helpful. LGTM ✅ 

I did quick run, everything seems to work, and I did quick run through different stories, especially module ones, didn't stumble upon anything broken or erroring, and storybook tests pass.

**Review by eugene-manuilov — APPROVED** [review]

**Review by eugene-manuilov — DISMISSED** [review]
@tofumatt, could you please fix merge conflicts?

**Comment by tofumatt** [comment]
@eugene-manuilov Done, should be good-to-go now 😄

**Comment by tofumatt** [comment]
(Since this was approved already just going to get this merged after it passes tests to prevent future merge conflicts and unblock #8826. 🙂 )

**Pull request merged**
